app/src/main/java/me/tatarka/bindingcollectionadapter/sample/ImmutableViewModel.kt

```diff
@@ -21,56 +17,91 @@ class ImmutableViewModel : ViewModel(), ImmutableListeners {
+                                                    checked = false
+                                            )
+                                        }
+                                // Pretend we are slow
+                                Thread.sleep(1000)
+                                callback.onResult(list, params.requestedStartPosition, 200)
                             }
-                        // Pretend we are slow
-                        Thread.sleep(1000)
-                        callback.onResult(list, params.requestedStartPosition, 200)
-                    }
-
-                    override fun loadRange(
-                        params: LoadRangeParams,
-                        callback: LoadRangeCallback<Any>
-                    ) {
-                        val list =
-                            (0 until params.loadSize).map {
-                                ImmutableItem(
-                                    index = it + params.startPosition,
-                                    checked = false
-                                )
+
+                            override fun loadRange(
+                                    params: LoadRangeParams,
+                                    callback: LoadRangeCallback<Any>
+                            ) {
+                                val list =
+                                        (0 until params.loadSize).map {
+                                            ImmutableItem(
+                                                    index = it + params.startPosition,
+                                                    checked = false
+                                            )
+                                        }
+                                // Pretend we are slow
+                                Thread.sleep(1000)
+                                callback.onResult(list)
                             }
-                        // Pretend we are slow
-                        Thread.sleep(1000)
-                        callback.onResult(list)
-                    }
-                }
-        }, 20).build()
+                        }
+            }, PAGE_SIZE).build()
+
+    val pagedListV3: LiveData<PagingData<Any>> = Pager<Int, Any>(PagingConfig(
+            pageSize = PAGE_SIZE,
+            maxSize = 100,
+            prefetchDistance = PAGE_SIZE * 2,
+            enablePlaceholders = false)) {
+        object : PagingSource<Int, Any>() {
+            override suspend fun load(params: LoadParams<Int>): LoadResult<Int, Any> {
+                val safeKey = params.key ?: 0
+                val list =
+                        (0 until params.loadSize).map {
+                            ImmutableItem(
+                                    index = it + safeKey,
+                                    checked = false
+                            )
+                        }
+                // Pretend we are slow
+                delay(1000)
+
```

Comment:
I recommend adding a 

```if (random.nextBoolean()) {
                    return LoadResult.Error(IllegalAccessError("Error plz try again"))
                }
``` 

to test out the error behavior.    I added it, along with the retry support but it is flaky.